tests/util/bft.py

```diff
@@ -289,60 +289,87 @@ def get_live_replicas(self):
         if expected is None:
             expected = lambda _: True
 
+        matching_view = None
+        nb_replicas_in_matching_view = 0
         try:
-            # Step 1: Wait for an agreed view that matches the "expected" predicate
-            last_agreed_view = None
-            replicas_in_expected_view = 0
-            with trio.fail_after(seconds=30):
-                while True:
-                    try:
-                        with trio.move_on_after(seconds=1):
-                            key = ['replica', 'Gauges', 'lastAgreedView']
-                            view = await self.metrics.get(replica_id, *key)
-                            if expected(view):
-                                last_agreed_view = view
-                                break
-                    except KeyError:
-                        # metrics not yet available, continue looping
-                        continue
+            matching_view = await self._wait_for_matching_agreed_view(replica_id, expected)
```

Comment:
Do we even need this check, since we check the expected view for n-f replicas in `_wait_for_active_view`. I guess the difference here is the predicate.

Reply:
Well, actually the predicate might say something like `lambda v: v > 0`.
So we don't know exactly which view number we are waiting for to be activated.
So, I've decided to first wait for a view that matches the predicated to be agreed. Only then we wait for this view to be activated. 

It's a sort of two-step process which helps preserve determinism and... one's sanity when facing view change :)

PS: this all started sounding very Lovecraftian, so I'll probably stop here :)))